src/adapter/src/coord/sequencer/inner/peek.rs

```diff
@@ -702,17 +766,20 @@ impl Coordinator {
                                 explain_ctx,
                             })
                         }
-                        PeekContext::Select => PeekStage::Finish(PeekStageFinish {
-                            validity,
-                            plan,
-                            id_bundle,
-                            target_replica,
-                            source_ids,
-                            determination,
-                            timestamp_context,
-                            optimizer,
-                            global_lir_plan,
-                        }),
+                        PeekContext::Select | PeekContext::CopyTo(_) => {
+                            PeekStage::Finish(PeekStageFinish {
```

Comment:
My idea is to have a new variant like `PeekStage::CopyTo` here that will end up calling a similar function as finish, but with only things needed for copy.

Reply:
Just updated